packages/cli/src/metadataGeneration/metadataGenerator.ts

```diff
@@ -113,7 +113,7 @@ export class MetadataGenerator {
   };
 
   private checkForPathParamSignatureDuplicates = (controllers: Tsoa.Controller[]) => {
-    const isParamRegExp = new RegExp('{|}|:');
+    const paramRegExp = new RegExp('{(\\w*)}|:(\\w+)', 'g');
```

Comment:
is there any regex pre defined for parsing the parameters?

Reply:
Ideally, for express/koa: Replace `{}` with `:` and feed that into [path-to-regexp](https://www.npmjs.com/package/path-to-regexp)

Reply:
I understand, I will push a new rewrite with simplifies and improves the check even more.. and kicks out "false flags" ;)